sdk/core/azure_core/src/policies/bearer_token_policy.rs

```diff
@@ -2,20 +2,27 @@
 // Licensed under the MIT License.
 
 use crate::{
-    credentials::TokenCredential,
+    credentials::{AccessToken, TokenCredential},
+    error::{Error, ErrorKind},
     headers::AUTHORIZATION,
     policies::{Policy, PolicyResult},
     Context, Request,
 };
+use async_lock::RwLock;
```

Comment:
Is this the desired RwLock? I modeled this off of the caching implementation in identity:
https://github.com/Azure/azure-sdk-for-rust/blob/feature/track2/sdk/identity/azure_identity/src/credentials/cache.rs#L4

Reply:
We don't want to depend on any particular async runtime, so you can't use `tokio` directly. This should be fine.